src/payment/checkout/Checkout.jsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-i18n';
+import { sendTrackEvent } from '@edx/frontend-analytics';
+
+import messages from './Checkout.messages';
+import { paymentSelector } from '../data/selectors';
+import { submitPayment } from '../data/actions';
+import AcceptedCardLogos from './assets/accepted-card-logos.png';
+
+import PaymentForm from './payment-form/PaymentForm';
+import FreeCheckoutOrderButton from './FreeCheckoutOrderButton';
+import { PayPalButton } from '../payment-methods/paypal';
+import { ApplePayButton } from '../payment-methods/apple-pay';
+import { ORDER_TYPES } from '../data/constants';
+
+class Checkout extends React.Component {
+  handleSubmitPayPal = () => {
+    // TO DO: after event parity, track data should be
+    // sent only if the payment is processed, not on click
+    // Check for ApplePay and Free Basket as well
+    sendTrackEvent(
+      'edx.bi.ecommerce.basket.payment_selected',
+      { type: 'click', category: 'checkout', paymentMethod: 'PayPal' },
+    );
+
+    this.props.submitPayment({ method: 'paypal' });
+  }
+
+  handleSubmitApplePay = () => {
+    // TO DO: after event parity, track data should be
+    // sent only if the payment is processed, not on click
+    // Check for PayPal and Free Basket as well
+    sendTrackEvent(
+      'edx.bi.ecommerce.basket.payment_selected',
+      { type: 'click', category: 'checkout', paymentMethod: 'Apple Pay' },
+    );
+
+    this.props.submitPayment({ method: 'apple-pay' });
+  }
+
+  handleSubmitCybersource = (formData) => {
```

Comment:
I think this is still miles ahead of where it was, but it _is_ a bit confusing to have `handleSubmitCybersource` and `handleSubmitCybersourceButtonClick` side by side... it's like... why are there two?  Which one is doing what?  

Is it possible to give the payment form a single click handler and call both from it?  Or do they get called at different times? (assuming I'll find out when I read `PaymentForm`...)

Reply:
The comment below `handleSubmitCybersource` and above `handleSubmitCybersourceButtonClick` says:
```
  // The payment form does client side validation that happens before
  // the submit handler above is fired. We send the tracking event here
  // on click of the submit button for parity with the old page.
```
Maybe there's a better way to describe the reason?